browser/autoplay/autoplay_permission_context_browsertest.cc

```diff
@@ -142,233 +143,125 @@ class AutoplayPermissionContextBrowserTest : public InProcessBrowserTest {
+  EXPECT_EQ(0, popup_prompt_factory->TotalRequestCount());
   WaitForPlaying(autoplay_method_url());
   EXPECT_TRUE(
       ExecuteScriptAndExtractString(contents(), kVideoPlayingDetect, &result));
   EXPECT_EQ(result, kVideoPlaying);
 
-  AskAutoplay();
-  popup_prompt_factory->ResetCounts();
   result.clear();
 
-  EXPECT_EQ(0, popup_prompt_factory->TotalRequestCount());
-
   NavigateToURLUntilLoadStop(autoplay_attr_url());
-  EXPECT_TRUE(popup_prompt_factory->RequestTypeSeen(
+  EXPECT_FALSE(popup_prompt_factory->is_visible());
+  EXPECT_FALSE(popup_prompt_factory->RequestTypeSeen(
       permissions::PermissionRequestType::PERMISSION_AUTOPLAY));
-  EXPECT_EQ(1, popup_prompt_factory->TotalRequestCount());
+  EXPECT_EQ(0, popup_prompt_factory->TotalRequestCount());
   WaitForPlaying(autoplay_attr_url());
   EXPECT_TRUE(
       ExecuteScriptAndExtractString(contents(), kVideoPlayingDetect, &result));
   EXPECT_EQ(result, kVideoPlaying);
 }
 
-// Click allow from promt (muted)
-IN_PROC_BROWSER_TEST_F(AutoplayPermissionContextBrowserTest, ClickAllowMuted) {
+// If content setting = BLOCK, ignore play() method call and do not show
+// permission prompt.
+IN_PROC_BROWSER_TEST_F(AutoplayPermissionContextBrowserTest,
+                       BlockAutoplayByMethod) {
   std::string result;
-  AskAutoplay();
+  BlockAutoplay();
   permissions::PermissionRequestManager* manager =
-      permissions::PermissionRequestManager::FromWebContents(contents());
+    permissions::PermissionRequestManager::FromWebContents(contents());
   auto popup_prompt_factory =
       std::make_unique<permissions::MockPermissionPromptFactory>(manager);
 
   EXPECT_EQ(0, popup_prompt_factory->TotalRequestCount());
-  popup_prompt_factory->set_response_type(
-      permissions::PermissionRequestManager::ACCEPT_ALL);
 
-  NavigateToURLUntilLoadStop(autoplay_method_muted_url());
-  EXPECT_TRUE(popup_prompt_factory->RequestTypeSeen(
+  NavigateToURLUntilLoadStop(autoplay_method_url());
+  EXPECT_FALSE(popup_prompt_factory->is_visible());
+  EXPECT_FALSE(popup_prompt_factory->RequestTypeSeen(
       permissions::PermissionRequestType::PERMISSION_AUTOPLAY));
-  EXPECT_EQ(1, popup_prompt_factory->TotalRequestCount());
-  WaitForPlaying(autoplay_method_muted_url());
-  EXPECT_TRUE(
-      ExecuteScriptAndExtractString(contents(), kVideoPlayingDetect, &result));
-  EXPECT_EQ(result, kVideoPlaying);
-
-  AskAutoplay();
-  popup_prompt_factory->ResetCounts();
-  result.clear();
-
   EXPECT_EQ(0, popup_prompt_factory->TotalRequestCount());
-
-  NavigateToURLUntilLoadStop(autoplay_attr_muted_url());
-  EXPECT_TRUE(popup_prompt_factory->RequestTypeSeen(
-      permissions::PermissionRequestType::PERMISSION_AUTOPLAY));
-  EXPECT_EQ(1, popup_prompt_factory->TotalRequestCount());
-  WaitForPlaying(autoplay_attr_muted_url());
   EXPECT_TRUE(
       ExecuteScriptAndExtractString(contents(), kVideoPlayingDetect, &result));
-  EXPECT_EQ(result, kVideoPlaying);
+  EXPECT_NE(result, kVideoPlaying);
```

Comment:
Could we extend this test case by verifying clicking `play()` can still play video after autoplay is blocked?

Reply:
nvm, that will not be counted as user action